packages/config-loader/src/lib/schema/filtering.ts

```diff
@@ -61,11 +61,25 @@ export function filterByVisibility(
       const arr = new Array<JsonValue>();
 
       for (const [index, value] of jsonVal.entries()) {
-        const out = transform(
-          value,
+        const hasVisibilityInIndex = visibilityByDataPath.get(
           `${visibilityPath}/${index}`,
-          `${filterPath}[${index}]`,
         );
+        let out;
+
+        if (typeof value !== 'object' && !hasVisibilityInIndex) {
```

Comment:
Wonder if we could tweak this logic a little to make it slightly easier to follow? If we figure out the visibilityPath first we can get away with just a single call to `transform`.

Reply:
@Rugvip yes of course. I made a change, what do you think?